Prevent mountpoint collision between ephemeral drive and shared drives

jdeamicis opened the pull request:
### Description of changes
* Prevent automatic mounting of ephemeral drives in case of a mountpoint collision between ephemeral drive and other shared storage devices.

### Tests
* Manually created clusters with and without colliding `/scratch` mount point with:
    * EBS
    * EFS
    * FSX
    * EBS-RAID

### Checklist
- [x] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [x] Check all commits' messages are clear, describing what and why vs how.
- [x] Make sure **to have added unit tests or integration tests** to cover the new/modified code.
- [x] Check if documentation is impacted by this change.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

jdeamicis committed:
Disable ephemeral mount if mountpoint clashes with shared file system mountpoint

Signed-off-by: Jacopo De Amicis <jdamicis@amazon.it>

lukeseawalker commented on cookbooks/aws-parallelcluster-config/recipes/base.rb:
```diff
@@ -27,16 +27,23 @@
+                   node['cluster']['efs_shared_dirs'].split(',') +\
+                   node['cluster']['fsx_shared_dirs'].split(',') +\
+                   [ node['cluster']['raid_shared_dir'] ]
+unless node['cluster']['ephemeral_dir'] == "/scratch" and shared_dir_array.include? "/scratch"
```
I think the fix should be more general than considering only "/scratch" as value, i.e.
this doesn't consider the case when fsx_shared_dirs = "/pluto" and ephemeral_dir = "/pluto". The CLI print just a warning but the issue is still there.

jdeamicis replied:
Addressed in the latest commit.

jdeamicis committed:
Set up kitchen tests for mountpoint collision for ephemeral and shared drives

Modify mountpoints for ephemeral drive to test mountpoint collision logic between ephemeral drive and shared drives.

Signed-off-by: Jacopo De Amicis <jdamicis@amazon.it>

lukeseawalker commented on cookbooks/aws-parallelcluster-config/recipes/base.rb:
```diff
@@ -27,16 +27,23 @@
-  supports restart: false
-  action :enable
-end
+# Disable mounting of ephemeral drive under /scratch if /scratch is used by another device
```
can you please rephrase this comment, PR title and description to reflect the fact that we aren't considering only the "/scratch" value?

jdeamicis replied:
Addressing in the next commit.

lukeseawalker commented on cookbooks/aws-parallelcluster-test/recipes/tests.rb:
```diff
@@ -363,9 +363,25 @@ module load intelmpi && mpirun --help | grep '#{node['cluster']['intelmpi']['kit
+    cwd Chef::Config[:file_cache_path]
+    user node['cluster']['cluster_user']
+    code <<-COLLISION
+      systemctl show setup-ephemeral.service | grep "ActiveState=inactive"
```
minor, you can use `systemctl show -p ActiveState setup-ephemeral.service` to restrict the output of the show command

jdeamicis replied:
Yes, that's better. Addressing in the next commit.

jdeamicis committed:
Address code review comments

Signed-off-by: Jacopo De Amicis <jdamicis@amazon.it>
